src/p2p/net_node.inl

```diff
@@ -847,6 +847,30 @@ namespace nodetool
   }
   //-----------------------------------------------------------------------------------
   template<class t_payload_net_handler>
+  bool node_server<t_payload_net_handler>::is_peer_used(const anchor_peerlist_entry& peer)
+  {
+    if(m_config.m_peer_id == peer.id) {
+        return true;//dont make connections to ourself
+    }
+
+    bool used = false;
+
+    m_net_server.get_config_object().foreach_connection([&](const p2p_connection_context& cntxt)
+    {
+      if(cntxt.peer_id == peer.id || (!cntxt.m_is_income && peer.adr.ip == cntxt.m_remote_ip && peer.adr.port == cntxt.m_remote_port))
```

Comment:
I don't think so. I just saw this check and thought that the expected behavior would be different. I don't think it will change much, but better to mention it. If you think it's fine, I'm fine with it too.